src/main/java/appeng/parts/p2p/PartP2PItems.java

```diff
@@ -147,17 +136,9 @@ private IInventory getOutputInv()
 
 			if( output == null )
 			{
-				if( te instanceof TileEntityChest )
-				{
-					output = Platform.GetChestInv( te );
-				}
-				else if( te instanceof ISidedInventory )
-				{
-					output = new WrapperMCISidedInventory( (ISidedInventory) te, this.getSide().getFacing().getOpposite() );
-				}
-				else if( te instanceof IInventory )
+				if( te.hasCapability( CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, this.getSide().getFacing().getOpposite() ) )
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1066)